### DISCUSSION TITLE
How to get clangd to work in AOSP?

### DISCUSSION THREAD

**Discussion by Hanmeim** [clangd/clangd#1818] [Q&A]
I'm working on THE aosp, which is quite a large project. So I'm trying to get clangd to work on the cpp files.
Google has already offered a way to generate compile_commands.json [here](https://android.googlesource.com/platform/build/soong/+/HEAD/docs/compdb.md)
But even with the compile_commands.json, clangd is still giving me tons of errors, and having difficulty finding the right header/definition/etc...
Is there any way to make it work?

**Comment by HighCommander4** [comment]
If you're looking for someone who has successfully set up the AOSP codebase specifically with clangd to give you pointers, you might have more luck asking in an AOSP-specific forum.

If you'd like help investigating the issues you're running into, please provide additional details, starting with:
 * [Clangd logs](https://clangd.llvm.org/troubleshooting#gathering-logs)
 * A description of what specifically isn't working (e.g. what file are you opening, what are you expecting to happen, what actually happens)
